src/main/java/org/jabref/gui/preferences/TableColumnsTabView.java

```diff
@@ -0,0 +1,145 @@
+
+import com.airhacks.afterburner.views.ViewLoader;
+import de.saxsys.mvvmfx.utils.validation.visualization.ControlsFxVisualizer;
+
+public class TableColumnsTabView extends AbstractPreferenceTabView implements PreferencesTab {
+
+    @FXML private TableView<TableColumnsItemModel> columnsList;
+    @FXML private TableColumn<TableColumnsItemModel, Field> nameColumn;
+    @FXML private TableColumn<TableColumnsItemModel, Field> actionsColumn;
+    @FXML private ComboBox<Field> addColumnName;
+    @FXML private CheckBox showFileColumn;
+    @FXML private CheckBox showUrlColumn;
+    @FXML private RadioButton urlFirst;
+    @FXML private RadioButton doiFirst;
+    @FXML private CheckBox showEPrintColumn;
+    @FXML private CheckBox specialFieldsEnable;
+    @FXML private Button specialFieldsHelp;
+    @FXML private RadioButton specialFieldsSyncKeywords;
+    @FXML private RadioButton specialFieldsSerialize;
+    @FXML private CheckBox extraFileColumnsEnable;
+
+    private ControlsFxVisualizer validationVisualizer = new ControlsFxVisualizer();
+
+    private final JabRefPreferences preferences;
+
+    public TableColumnsTabView(JabRefPreferences preferences) {
+        this.preferences = preferences;
+
+        ViewLoader.view(this)
+                .root(this)
+                .load();
+    }
+
+    @Override
+    public String getTabName() {
+        return Localization.lang("Entry table columns");
+    }
+
+    public void initialize() {
+        this.viewModel = new TableColumnsTabViewModel(dialogService, preferences);
+
+        setupTable();
+        setupBindings();
+
+        ActionFactory actionFactory = new ActionFactory(preferences.getKeyBindingRepository());
+        actionFactory.configureIconButton(StandardActions.HELP_SPECIAL_FIELDS, new HelpAction(HelpFile.SPECIAL_FIELDS), specialFieldsHelp);
+    }
+
+    private void setupTable() {
+        nameColumn.setSortable(false);
+        nameColumn.setReorderable(false);
+        nameColumn.setCellValueFactory(cellData -> cellData.getValue().fieldProperty());
+        new ValueTableCellFactory<TableColumnsItemModel, Field>().withText(FieldsUtil::getNameWithType).install(nameColumn);
+
+        actionsColumn.setSortable(false);
+        actionsColumn.setReorderable(false);
+        actionsColumn.setCellValueFactory(cellData -> cellData.getValue().fieldProperty());
+        new ValueTableCellFactory<TableColumnsItemModel, Field>()
+                .withGraphic(item -> IconTheme.JabRefIcons.DELETE_ENTRY.getGraphicNode())
+                .withTooltip(name -> Localization.lang("Remove column") + " " + name.getDisplayName())
+                .withOnMouseClickedEvent(item -> evt ->
+                        ((TableColumnsTabViewModel) viewModel).removeColumn(columnsList.getFocusModel().getFocusedItem()))
+                .install(actionsColumn);
+
+        ((TableColumnsTabViewModel) viewModel).selectedColumnModelProperty().setValue(columnsList.getSelectionModel());
```

Comment:
For the next PR: you can get rid off these castings as follows:
- Add a generic type argument: `AbstractPreferenceTabView<T extends PreferenceTabViewModel>`
- Change the variable to `protected T viewModel;`
- Change declaration to `TableColumnsTabView extends AbstractPreferenceTabView<TableColumnsTabViewModel>`